autograder-core/src/main/java/de/firemage/autograder/core/CodePosition.java

```diff
@@ -55,7 +57,14 @@ public static CodePosition fromSourcePosition(
         );
     }
 
-    public String readString() {
+
+    @Override
+    public Path path() {
+        return this.file.toPath();
```

Comment:
Note that the path won't always be a real path. Some things are only in memory (I am not sure if I used somewhere other than the tests)